netCDF4/utils.py

```diff
@@ -269,7 +269,7 @@ def _StartCountStride(elem, shape, dimensions=None, grp=None, datashape=None,\
                 ee = range(start,stop,step)
             except ValueError: # start, stop or step is not valid for a range
                 ee = False
-            if ee and len(e) == len(ee) and (e == np.arange(start,stop,step)).all():
+            if ee and e[1]-e[0]==1 and len(e) == len(ee) and (e == np.arange(start,stop,step)).all():
```

Comment:
How about 

            if ee and len(e) == len(ee) and (e == np.arange(start,stop,step)).all():
                if step != 1 and len(e) < 1000: # don't convert to slice
                    newElem.append(e)
                else:  # convert to slice
                    newElem.append(slice(start,stop,step))